examples/platforms/nrf528xx/src/radio.c

```diff
@@ -695,9 +708,15 @@ otError otPlatRadioGetTransmitPower(otInstance *aInstance, int8_t *aPower)
 otError otPlatRadioSetTransmitPower(otInstance *aInstance, int8_t aPower)
 {
     OT_UNUSED_VARIABLE(aInstance);
+    int8_t  power;
+    uint8_t channel = nrf_802154_channel_get();
 
-    sDefaultTxPower = aPower;
-    nrf_802154_tx_power_set(aPower);
+    sDefaultTxPower   = aPower;
+    sPowerInitialized = true;
+    power             = aPower < sMaxTxPowerTable[channel - OT_RADIO_2P4GHZ_OQPSK_CHANNEL_MIN]
```

Comment:
Sanity check for `channel` before use or use `nrf5GetChannelMaxTransmitPower` instead?